cli/src/description_util.rs

```diff
@@ -31,12 +43,138 @@ JJ: Lines starting with "JJ: " (like this one) will be removed.
         settings,
     )?;
 
-    // Normalize line ending, remove leading and trailing blank lines.
-    let description = description
-        .lines()
-        .filter(|line| !line.starts_with("JJ: "))
-        .join("\n");
-    Ok(text_util::complete_newline(description.trim_matches('\n')))
+    Ok(cleanup_description(&description))
+}
+
+#[derive(Debug)]
+pub struct EditMultipleDescriptionsResult {
+    /// The parsed, formatted descriptions.
+    pub descriptions: HashMap<CommitId, String>,
+    /// Commit IDs that were expected while parsing the edited messages, but
+    /// which were not found.
+    pub missing: Vec<String>,
+    /// Commit IDs that were found multiple times while parsing the edited
+    /// messages.
+    pub duplicates: Vec<String>,
+    /// Commit IDs that were found while parsing the edited messages, but which
+    /// were not originally being edited.
+    pub unexpected: Vec<String>,
+}
+
+/// Edits the descriptions of the given commits in a single editor session.
+pub fn edit_multiple_descriptions(
+    tx: &mut WorkspaceCommandTransaction,
+    repo: &ReadonlyRepo,
+    commits: &[&Commit],
+    settings: &UserSettings,
+) -> Result<EditMultipleDescriptionsResult, CommandError> {
+    let mut commits_map = HashMap::new();
+    let mut output_chunks = Vec::new();
+
+    for &commit in commits.iter() {
+        let commit_hash = short_commit_hash(commit.id());
+        if commits.len() > 1 {
```

Comment:
It's probably better to let caller use `edit_description()` if `commits.len() == 1`, because:
- no `JJ: describe` header nor parsing is needed
- different description template has to be chosen for multiple edits (if we add support for embedded diffs without `JJ: ` prefixes.)
- the caller might have different behavior (e.g. `describe -m M x y ..` might be better to error out or open editor with all commit messages filled with `M` by default.)